README.md

```diff
@@ -1,2 +1,66 @@
-# puppetlabs-ruby_task_helper
-A ruby library for running tasks
+# Ruby Task Helper
+
+A Ruby helper library for writing [Puppet tasks](https://puppet.com/docs/bolt/latest/writing_tasks.html). It provides a class that handles error generation, simplifies JSON input and output, and makes testing your task easier. It requires Bolt >= 1.1 and Puppet Enterprise >= 2019.0.
+
+#### Table of Contents
+
+1. [Description](#description)
+1. [Requirements](#requirements)
+1. [Setup](#setup)
+1. [Usage](#usage)
+
+## Description
+
+This library handles parsing JSON input, serializing the result as JSON output, and producing a formatted error message for errors.
+
+## Requirements
+
+This library works with Ruby 2.3 and later.
+
+## Setup
+
+To use this library, include this module in a [Puppetfile]():
+
+```
+mod puppetlabs-ruby_task_helper
+```
+
+Add it to your [task metadata](https://puppet.com/docs/bolt/latest/writing_tasks.html#concept-677)
+```
+{
+  "files": ["ruby_task_helper/lib/task_helper.rb"]
```

Comment:
I ended up including `input_method` as well as a way of clarifying that the helper expects input on `stdin`.